server/src/main/java/org/elasticsearch/action/support/master/TransportMasterNodeAction.java

```diff
@@ -115,6 +132,15 @@ protected void doExecute(final Request request, ActionListener<Response> listene
 
         public void start() {
             ClusterState state = clusterService.state();
+            if (state == null) {
+                WaitForInitialState waitForState = new WaitForInitialState(this);
+                clusterService.addListener(waitForState);
+                // protect against race between state null check and listener registration
+                if (clusterService.state() != null) {
+                    waitForState.clusterChanged(null);
```

Comment:
shouldn't this be:
```suggestion
                    waitForState.clusterChanged(clusterService.state());
```
?

Reply:
`clusterChanged` would take a `ClusterChangedEvent`. It is not used and I didn't want to create an instance - the listener is only used as a notification to re-try running `start()` again.

`start` tries to get the current state again, and if it is still not available it would repeat.

Reply:
ah right, my bad 👍